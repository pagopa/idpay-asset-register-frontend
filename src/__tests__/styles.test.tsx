import {
    genericContainerStyle,
    pagesTableContainerStyle,
    pagesFiltersFormContainerStyle,
} from '../styles';

describe('Style objects', () => {
    test('should match the snapshot', () => {
        expect(genericContainerStyle).toMatchSnapshot();
        expect(pagesTableContainerStyle).toMatchSnapshot();
        expect(pagesFiltersFormContainerStyle).toMatchSnapshot();
    });

    test('genericContainerStyle has correct properties', () => {
        expect(genericContainerStyle).toEqual(
            expect.objectContaining({
                display: 'grid',
                width: '100%',
                gridTemplateColumns: 'repeat(12, 1fr)',
            })
        );
    });

    test('pagesTableContainerStyle has correct properties', () => {
        expect(pagesTableContainerStyle).toEqual(
            expect.objectContaining({
                display: 'grid',
                width: '100%',
                gridTemplateColumns: 'repeat(12, 1fr)',
                alignItems: 'center',
            })
        );
    });

    test('pagesFiltersFormContainerStyle has correct properties', () => {
        expect(pagesFiltersFormContainerStyle).toEqual(
            expect.objectContaining({
                display: 'grid',
                width: '100%',
                gridTemplateColumns: 'repeat(12, 1fr)',
                alignItems: 'baseline',
                gap: 2,
                mb: 4,
            })
        );
    });
});