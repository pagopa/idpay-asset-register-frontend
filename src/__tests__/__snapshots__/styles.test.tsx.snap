// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Style objects should match the snapshot 1`] = `
Object {
  "display": "grid",
  "gridTemplateColumns": "repeat(12, 1fr)",
  "width": "100%",
}
`;

exports[`Style objects should match the snapshot 2`] = `
Object {
  "alignItems": "center",
  "display": "grid",
  "gridTemplateColumns": "repeat(12, 1fr)",
  "width": "100%",
}
`;

exports[`Style objects should match the snapshot 3`] = `
Object {
  "alignItems": "baseline",
  "display": "grid",
  "gap": 2,
  "gridTemplateColumns": "repeat(12, 1fr)",
  "mb": 4,
  "width": "100%",
}
`;
